### PR TITLE
Port AzurePowerShell V2 to master

### PR DESCRIPTION
**Pull request by vincentdass**

### CODE REVIEW

File: make-options.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
         "AzureMonitorV0",
         "AzureMysqlDeploymentV1",
         "AzureNLBManagementV1",
+		"AzurePowerShellV2",
```
Comment (asranja) [inline make-options.json]: Indentation.

File: Tasks/AzurePowerShellV2/AzurePowerShell.ps1 [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+catch
+{
+   $error = $_.Exception.Message
+   Write-Verbose "Unable to get the authScheme $error" 
+}
+
+if ($endpoint.Auth.parameters.authenticationType -ieq "spnCertificate")
+{
+	throw (Get-VstsLocString -Key CertificateEndpointNotSupported)
+}
```
Comment (asranja) [inline Tasks/AzurePowerShellV2/AzurePowerShell.ps1]: We dont need this. AzurePowerShellV2 will automatically start using the latest common module which has support for certificate based SPN.

File: Tasks/AzurePowerShellV2/AzurePowerShell.ps1 [hunk 3]
```diff
@@ -0,0 +1,147 @@
+Trace-VstsEnteringInvocation $MyInvocation
```
Comment (asranja) [inline Tasks/AzurePowerShellV2/AzurePowerShell.ps1]: Add TPN for openssl.

Comment (vincentdass) [reply Tasks/AzurePowerShellV2/AzurePowerShell.ps1]: done. thanks.

**Pull request merged**